#! /bin/sh /usr/share/dpatch/dpatch-run
## datadir.patch by  Matthias Klumpp <matthias@nlinux.org>
##
## DP: Correct data search path in smile

@DPATCH@
Index: smile-0.9.12/mainfrm.cpp
===================================================================
--- smile-0.9.12.orig/mainfrm.cpp	2009-08-25 18:38:13.000000000 +0200
+++ smile-0.9.12/mainfrm.cpp	2009-08-27 14:59:33.000000000 +0200
@@ -168,6 +168,13 @@
 
 mainfrm::mainfrm(QWidget *parent) : QMainWindow(parent)
 {
+	resourceDirPath = "/usr/share/smile";
+	QDir *directory = new QDir();
+	if (!directory->exists(resourceDirPath))
+	{
+	  resourceDirPath = qApp->applicationDirPath();
+	}
+
 	QPixmap yopixmap(":/new/prefix2/Interface/Theme/splash_smile.jpg");
 	QSplashScreen splashy(yopixmap);
 	Qt::Alignment sypal = Qt::AlignRight | Qt::AlignBottom;
@@ -183,13 +190,13 @@
 		QString wlpack = QLocale::system().name();
 		qDebug() << "LOADING LANGUAGE ... smile_" + wlpack.left(2);
 		QTranslator *qtTranslator = new QTranslator(this);
-		if (qtTranslator->load("smile_" + wlpack.left(2), qApp->applicationDirPath()) == true )
+		if (qtTranslator->load("smile_" + wlpack.left(2), resourceDirPath) == true )
 		{
 			qApp->installTranslator(qtTranslator);
 		}
 		else
 		{
-			if (qtTranslator->load("smile_en", qApp->applicationDirPath()) == true )
+			if (qtTranslator->load("smile_en", resourceDirPath) == true )
 			{
 				qApp->installTranslator(qtTranslator);
 			}
@@ -1317,13 +1324,13 @@
 				program = "xcopy";
 				arguments << "/e";
 				arguments << "/s";
-				arguments << QApplication::applicationDirPath() + "\BIB_ManSlide\\";
+				arguments << resourceDirPath + "\BIB_ManSlide\\";
 				arguments << tempfolder->text();
 			#else
 				program = "cp";
 				arguments << "-r";
 				arguments << "-f";
-				arguments << QApplication::applicationDirPath() + "/BIB_ManSlide/";
+				arguments << resourceDirPath + "/BIB_ManSlide/";
 				arguments << tempfolder->text();
 			#endif
 			ispreview = 4;
Index: smile-0.9.12/mainfrm.h
===================================================================
--- smile-0.9.12.orig/mainfrm.h	2009-08-25 11:38:46.000000000 +0200
+++ smile-0.9.12/mainfrm.h	2009-08-27 14:59:33.000000000 +0200
@@ -79,6 +79,7 @@
 		QString wglob_str_1;
 		QString wg_library;
 		QString wg_save_filename;
+		QString resourceDirPath;
 		int size_dia_x;
 		int size_dia_y;
 		int etat_lecture;
Index: smile-0.9.12/helpfrm.cpp
===================================================================
--- smile-0.9.12.orig/helpfrm.cpp	2009-08-18 15:12:44.000000000 +0200
+++ smile-0.9.12/helpfrm.cpp	2009-08-27 14:59:33.000000000 +0200
@@ -26,11 +26,11 @@
 			qDebug() << "LOADING HELP LANGUAGE ... " + wlpack.left(2);
 			if ( wlpack.left(2) == "fr" )
 			{
-				webhelp->load(QUrl(wg_tmp_002 + "/BIB_ManSlide/Help/doc_fr.html"));
+				webhelp->load(QUrl("/usr/share/smile/BIB_ManSlide/Help/doc_fr.html"));
 			}
 			else
 			{
-				webhelp->load(QUrl(wg_tmp_002 + "/BIB_ManSlide/Help/doc_en.html"));
+				webhelp->load(QUrl("/usr/share/smile/BIB_ManSlide/Help/doc_en.html"));
 			}
 		}
 		configfile.close();
